piptools/_compat/pip_compat.py

```diff
@@ -36,6 +38,26 @@ def parse_requirements(
         yield install_req_from_parsed_requirement(parsed_req, isolated=isolated)
 
 
+def install_req_from_line(
+    name: str,
+    comes_from: str | InstallRequirement | None = None,
+    base_package: str | None = None,
+    base_dir: str | None = None,
+) -> InstallRequirement:
+    parts = parse_req_from_line(name, comes_from)
+    if base_package and base_dir and parts.requirement.name == base_package:
+        name = name.replace(base_package, base_dir, 1)
+        parts = parse_req_from_line(name, comes_from)
```

Comment:
This looks better, but I still have the feeling that this can be improved. I'll try to come up with something. Thanks!

Reply:
After taking closer look, I didn't find better solution. Let's keep it as is.